src/std/foreign.ss

```diff
@@ -43,10 +44,172 @@
+                                           nonnull-UTF-8-string UTF-16-string
+                                           nonnull-UTF16-string))
+               (string-compat-required? (let loop ((m members))
+                                          (cond
+                                           ((null? m) #f)
+                                           ((member (cdr (car m)) string-types) #t)
+                                           (else (loop (cdr m))))))
+               (string-setter-body (lambda (member-name)
+                                     (let ((m (string-append "___arg1->" member-name)))
+                                       (string-append
+                                        "if(" m " == NULL)" "\n"
+                                        m "= strdup(___arg2);" "\n"
+                                        "else if (strcmp(" m ", ___arg2) != 0) {" "\n"
+                                        "free(" m ");" "\n"
+                                        m "= strdup(___arg2);" "\n"
+                                        "}" "\n"
+                                        "___return;" "\n"))))
+               (default-free-body (and string-compat-required?
+                                       (string-append
+                                        "___SCMOBJ " struct-str "_ffi_free (void *ptr) {" "\n"
+                                        "struct " struct-str " *obj = (struct " struct-str "*) ptr;" "\n"
+                                        (apply string-append
+                                          (map (lambda (m)
+                                                 (case (cdr m)
+                                                   ((char-string)
+                                                    (let ((mem-name (symbol->string (car m))))
+                                                      (string-append "if(obj->" mem-name ") " 
+                                                                     "free(obj->" mem-name ");" "\n")))
+                                                                           (else "")))
+                                               members))
+                                        "free(obj);" "\n"
+                                        "return ___FIX (___NO_ERR);" "\n"
+                                        "}"
+                                        )))
+               (release-function (or release-function
+                                     (if string-compat-required?
+                                       (string-append struct-str "_ffi_free")
+                                       "ffi_free")))
+               (string-compat-types (if string-compat-required?
+                                      `((c-declare ,default-free-body)
+                                        (c-define-type ,shallow-ptr
+                                              (pointer ,struct (,struct-ptr) "ffi_free"))))))
```

Comment:
the `else` branch needs to return a list here, otherwise it will be just void and fail in the append.

Reply:
or splice if we get rid of the append.